### ISSUE TITLE
Fails when building using microsoft/aspnetcore-build docker image

### ISSUE THREAD

**Issue by jasonchester** [AArnott/Nerdbank.GitVersioning#177]
Docker file
```
# Sample contents of Dockerfile
# Stage 1
FROM microsoft/aspnetcore-build AS builder
WORKDIR /source

# caches restore result by copying csproj file separately
COPY *.csproj .
RUN dotnet restore

# copies the rest of your code
COPY . .
RUN dotnet publish --output /app/ --configuration Release

# Stage 2
FROM microsoft/aspnetcore

WORKDIR /app
COPY --from=builder /app .
ENTRYPOINT ["dotnet", "WebApi.dll"]
```

Relevant Output

```
Step 6/16 : RUN dotnet publish --output /app/ --configuration Release
 ---> Running in e3e69838076b
Microsoft (R) Build Engine version 15.6.82.30579 for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

  Restore completed in 91.92 ms for /source/WebAPI.csproj.
  Restore completed in 91.9 ms for /source/WebAPI.csproj.
  Restore completed in 105.77 ms for /source/WebAPI.csproj.
  Restore completed in 43.17 ms for /source/WebAPI.csproj.
  Restore completed in 23.07 ms for /source/WebAPI.csproj.
/root/.nuget/packages/nerdbank.gitversioning/2.1.23/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018: The "Nerdbank.GitVersioning.Tasks.GetBuildVersion" task failed unexpectedly. [/source/Brandmuscle.LocationData.WebAPI.csproj]
/root/.nuget/packages/nerdbank.gitversioning/2.1.23/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018: System.TypeInitializationException: The type initializer for 'LibGit2Sharp.Core.NativeMethods' threw an exception. ---> System.DllNotFoundException: Unable to load DLL '/root/.nuget/packages/nerdbank.gitversioning/2.1.23/build/MSBuildFull/lib/linux/x86_64/libgit2-1196807.so': The specified module or one of its dependencies could not be found. [/source/Brandmuscle.LocationData.WebAPI.csproj]
/root/.nuget/packages/nerdbank.gitversioning/2.1.23/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:  (Exception from HRESULT: 0x8007007E) [/source/Brandmuscle.LocationData.WebAPI.csproj]
/root/.nuget/packages/nerdbank.gitversioning/2.1.23/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at System.Runtime.Loader.AssemblyLoadContext.InternalLoadUnmanagedDllFromPath(String unmanagedDllPath) [/source/Brandmuscle.LocationData.WebAPI.csproj]
/root/.nuget/packages/nerdbank.gitversioning/2.1.23/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at MSBuildExtensionTask.ContextAwareTask.CustomAssemblyLoader.LoadUnmanagedDll(String unmanagedDllName) [/source/Brandmuscle.LocationData.WebAPI.csproj]
/root/.nuget/packages/nerdbank.gitversioning/2.1.23/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at System.Runtime.Loader.AssemblyLoadContext.ResolveUnmanagedDll(String unmanagedDllName, IntPtr gchManagedAssemblyLoadContext) [/source/Brandmuscle.LocationData.WebAPI.csproj]
/root/.nuget/packages/nerdbank.gitversioning/2.1.23/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at LibGit2Sharp.Core.NativeMethods.git_libgit2_init() [/source/Brandmuscle.LocationData.WebAPI.csproj]
/root/.nuget/packages/nerdbank.gitversioning/2.1.23/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at LibGit2Sharp.Core.NativeMethods.LibraryLifetimeObject..ctor() [/source/Brandmuscle.LocationData.WebAPI.csproj]
/root/.nuget/packages/nerdbank.gitversioning/2.1.23/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at LibGit2Sharp.Core.NativeMethods..cctor() [/source/Brandmuscle.LocationData.WebAPI.csproj]
/root/.nuget/packages/nerdbank.gitversioning/2.1.23/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    --- End of inner exception stack trace --- [/source/Brandmuscle.LocationData.WebAPI.csproj]
/root/.nuget/packages/nerdbank.gitversioning/2.1.23/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at LibGit2Sharp.Core.NativeMethods.git_libgit2_opts(Int32 option, UInt32 level, String path) [/source/Brandmuscle.LocationData.WebAPI.csproj]
/root/.nuget/packages/nerdbank.gitversioning/2.1.23/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at LibGit2Sharp.GlobalSettings.SetConfigSearchPaths(ConfigurationLevel level, String[] paths) [/source/Brandmuscle.LocationData.WebAPI.csproj]
/root/.nuget/packages/nerdbank.gitversioning/2.1.23/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at Nerdbank.GitVersioning.VersionOracle.OpenGitRepo(String repoRoot) [/source/Brandmuscle.LocationData.WebAPI.csproj]
/root/.nuget/packages/nerdbank.gitversioning/2.1.23/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at Nerdbank.GitVersioning.VersionOracle.Create(String projectDirectory, String gitRepoDirectory, ICloudBuild cloudBuild, Nullable`1 overrideBuildNumberOffset, String projectPathRelativeToGitRepoRoot) [/source/Brandmuscle.LocationData.WebAPI.csproj]
/root/.nuget/packages/nerdbank.gitversioning/2.1.23/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at Nerdbank.GitVersioning.Tasks.GetBuildVersion.ExecuteInner() [/source/Brandmuscle.LocationData.WebAPI.csproj]
The command '/bin/sh -c dotnet publish --output /app/ --configuration Release' returned a non-zero code: 1
```

**Comment by AArnott** [comment]
Hmmm... this is an interesting one. The file is clearly in the package. Why can't it find it?
But I also notice that although you're using `dotnet build`, it's nevertheless using the `MSBuildFull` path, which looks like a regression (in `dotnet build`?).

**Comment by AArnott** [comment]
Well your issue is at least on a.net core supported flavor of Linux which is more than I can say for the other issue.

**Comment by ekwus** [comment]
Looking at another docker file for an aspnercore build [here](https://github.com/Franklin89/gitversion-aspnetcore-build/blob/master/Dockerfile) shows the libgit2 library being installed.

Do you think the environment is missing components that NuGet can't pull in?

I've tried adding the RUN apt-get install libgit2-24 from the other docker file but it doesn't solve the issue. Do you know what version of libgit2 the NuGet package depends on or is it all statically linked and I'm barking up the wrong tree?

Cheers

Dave

**Comment by ekwus** [comment]
Ok

So I've tracked this down to the version of OpenSSL in the image is 1.1.0 but the version of libgit2 being used depends on OpenSSL 1.0.0.

I tried various combinations of symbolic linking of the libssl and libcrypto files that are missing but all failed.

Deleting the packages libgit2-1196807.so and adding a symlink to libgit2-24 (which I installed using apt-get install libgit2-24) has resolve the issue, albeit its a big hack!

Cheers

Dave
